backend/src/modules/catalog/typedefs/catalog.ts

```diff
@@ -1,142 +1,319 @@
 import { gql } from "graphql-tag";
 
 export default gql`
-type Query {
+  type Query {
     course(subject: String!, courseNumber: String!, term: TermInput): Course
-    class(subject: String!, courseNumber: String!, term: TermInput!, classNumber: String!): Class
-    section(subject: String!, courseNumber: String!, term: TermInput!, classNumber: String!, sectionNumber: String!): Section
+    class(
+      subject: String!
+      courseNumber: String!
+      term: TermInput!
+      classNumber: String!
+    ): Class
+    section(
+      subject: String!
+      courseNumber: String!
+      term: TermInput!
+      classNumber: String!
+      sectionNumber: String!
+    ): Section
 
     """
-    Get info about all courses and their corresponding classes for a given semester. 
-    
+    Get info about all courses and their corresponding classes for a given semester.
+
     Used primarily in the catalog page.
     """
-    catalog(term: TermInput!): [CatalogItem]
+    catalog(term: TermInput!): [Course!]
```

Comment:
The reason why I had a separate item from `Course` here originally was to ensure that it was performant enough. The catalog query was already quite slow, although proper indexing may help that a lot, but just be cautious about this, especially if a random person tries to make a nested query (e.g. `{ catalog { classes { sections { number } } } }`), that could really put a load on the server and/or database

Reply:
@mathhulk might have more to say but he said he's planning to add a max depth on the queries.

Reply:
If so please make an issue or something for that and make sure it gets done before this is released to the public!